pkg/clients/cloudsql/cloudsql.go

```diff
@@ -283,10 +296,17 @@ func LateInitializeSpec(spec *v1beta1.CloudSQLInstanceParameters, in sqladmin.Da
 
 // IsUpToDate checks whether current state is up-to-date compared to the given
 // set of parameters.
-func IsUpToDate(in *v1beta1.CloudSQLInstanceParameters, currentState sqladmin.DatabaseInstance) bool {
-	currentParams := &v1beta1.CloudSQLInstanceParameters{}
-	LateInitializeSpec(currentParams, currentState)
-	return cmp.Equal(in, currentParams, cmpopts.IgnoreInterfaces(struct{ resource.AttributeReferencer }{}))
+func IsUpToDate(name string, in *v1beta1.CloudSQLInstanceParameters, currentState *sqladmin.DatabaseInstance) (bool, error) {
+	genState, err := copystructure.Copy(currentState)
+	if err != nil {
+		return false, err
+	}
+	_, ok := genState.(*sqladmin.DatabaseInstance)
+	if !ok {
+		return false, errors.New("unable to type case deepcopy to *sqladmin.DatabaseInstance")
+	}
```

Comment:
Two things here:

* Is `false` the right value to return here if we see an error? Perhaps it's safer to return `true` (i.e. don't try to update the resource) in the error case.
* This error will bubble up to the user; this might be a little too much detail for them.